docs/getting-started/dockstore-tools.rst

```diff
@@ -8,18 +8,102 @@ Register a Tool on Dockstore
+
+.. image:: /assets/images/docs/add-tool-button.png
+   :width: 40 %
+
+.. image:: /assets/images/docs/register-tool-github-apps.png
+   :width: 40 %
+
+.. figure:: /assets/images/docs/gh-app-reg-tool.png
+   :width: 70 %
+
+   Install our GitHub App on either all repositories in an organization or on specific repositories
+
+Once you've installed our GitHub app on a repository or organization, you'll need to add a ``/.dockstore.yml`` file to
+the root directory of a branch of the repository that contains your tool. This file contains information like
+tool path, test parameter file, workflow name, etc. When a push is made on GitHub to a branch
+with a ``/.dockstore.yml``, Dockstore will add that branch to the corresponding tool on Dockstore. If the
+tool doesn't already exist on Dockstore, one will be created. Note that a single ``/.dockstore.yml`` file can describe multiple tools, if all of those tools are in the same repository.
+
+Below is a simple example of a ``/.dockstore.yml`` file
+for an alignment workflow to show you how easy it is to use. If you are interested in using this method, please see the
+complete documentation at the :doc:`Dockstore GitHub Apps <github-apps/github-apps>` page. All paths in the file must be absolute.
+
+.. code:: yaml
+
+   version: 1.2
+   tools:
+      - subclass: CWL
+        primaryDescriptorPath: /aligner.cwl
+        testParameterFiles:
+        - /test/aligner.cwl.json
+
+If you had our GitHub App installed on the repository ``myorg/alignments`` and then add the above ``/.dockstore.yml`` to the **develop** branch,
+the following would occur.
+
+* A **CWL** tool with the ID ``github.com/myorg/alignments`` will be created on Dockstore
+* The version **develop** is added to the tool ``github.com/myorg/alignments``
+* The version has the primary descriptor file set to ``/aligner.cwl``
+* The version has one test parameter file: ``/test/aligner.cwl.json``
+
+Now that your tool has been added, any time there is a push to a branch on GitHub for this repository that has a ``/.dockstore.yml``,
+it is automatically updated on Dockstore! Anytime there is a deletion of a branch on GitHub that has a ``/.dockstore.yml``, the version is
+removed from Dockstore.
+
+Traditional Registration
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. important:: The following methods are NOT recommended and should only be used if your tool descriptor files are registered on BitBucket or GitLab.
+
+.. important:: These methods require that you have created or have permissions to a Docker image. If that is not the case, consider reading our creating a Docker image or switching to GitHub as your code hosting platform.
+
+To use the following registration options, you need have your ``Dockerfile`` created in the :doc:`Getting Started with Docker <getting-started-with-docker>` tutorial, ``Dockstore.cwl`` in GitHub, have setup Quay.io to automatically build
```

Comment:
> Doesn't mention WDL (as long as we have at least one way to register WDL tools, WDL needs a mention here, because this section is for a group of registration options)

This is for the "traditional" registration methods (non github app tool), so is it possible to register a WDL tool with them? Yes, but we are trying to get users to not do it anymore so I am against even mentioning it anymore. I know it will still be possible in 1.12, but I just don't think we should mention it. Maybe I'm being too aggressive

But otherwise, your suggestion is much better. Thanks!